lucet-module-data/src/module.rs

```diff
@@ -0,0 +1,28 @@
+use crate::functions::FunctionSpec;
+use crate::module_data::ModuleData;
+use crate::tables::TableElement;
+
+pub const LUCET_MODULE_SYM: &str = "lucet_module";
+
+/// Module is the exposed structure that contains all the data backing a Lucet-compiled object.
+#[derive(Debug)]
+pub struct Module<'a> {
+    pub module_data: ModuleData<'a>,
+    pub tables: &'a [&'a [TableElement]],
+    pub function_manifest: &'a [FunctionSpec],
+}
+
+/// NativeData is a serialization-friendly form of Module, in that `ModuleData` is serialized, with
+/// the `module_data_*` fields here referring to that serialized data, while `tables_*` and
+/// `function_manifest_*` refer to the actual tables and function manifest data written in the
+/// binary.
+#[repr(C)]
+#[derive(Debug)]
+pub struct NativeData {
```

Comment:
:bike: :derelict_house: Maybe call this `SerializedModule` or `FlattenedModule`, replacing `Module` potentially with another name, if we decide to change that? `Native` makes me think x86-64 or just "non-wasm" in the context of our project.

Reply:
Definitely agree that `Serialized_` for whatever `Module` does or doesn't get renamed to is a better name, either way